crates/iroha_core/src/sumeragi/main_loop.rs

```diff
@@ -872,67 +853,56 @@ impl Sumeragi {
             .max_transactions
             .try_into()
             .expect("INTERNAL BUG: transactions in block exceed usize::MAX");
-        let block_time = if self.topology.view_change_index() > 0 {
-            Duration::from_secs(0)
-        } else {
-            state.world.view().parameters.sumeragi.block_time()
-        };
+
         let tx_cache_full = self.transaction_cache.len() >= max_transactions.get();
+        let view_change_in_progress = self.topology.view_change_index() > 0;
+        let block_time = state.world.view().parameters.sumeragi.block_time();
         let deadline_reached = self.round_start_time.elapsed() > block_time;
         let tx_cache_non_empty = !self.transaction_cache.is_empty();
 
-        if tx_cache_full || (deadline_reached && tx_cache_non_empty) {
+        if tx_cache_full || tx_cache_non_empty && (view_change_in_progress || deadline_reached) {
```

Comment:
Yeah, new leader should immediately create a block after view changes

Reply:
take a look [here](https://github.com/hyperledger/iroha/pull/5140/files#diff-38f46ad9bced8401be3c32f91d9b1b17265a043d3059b846e58235afe06a42c6R1109-R1111) also